File: Julia-files/Getting Data and Linear Models.jl
using CSV, DataFrames, Statistics, GLM, Plots

# --- Read CSV File ---
data = CSV.read("data.csv", DataFrame)

println(first(data, 5))
println(describe(data))


# --- Fit Linear Regression Model ---
model = lm(@formula(y ~ x), data)

println(model)


# --- Predictions ---
newx = DataFrame(x = [2, 5, 8])
predictions = predict(model, newx)

println(predictions)


# --- Plot with Regression Line ---
scatter(data.x, data.y,
    xlabel = "x",
    ylabel = "y",
    title = "Linear Regression",
    legend = false,
    markersize = 5)

xrange = range(minimum(data.x), maximum(data.x), length=100)
yrange = predict(model, DataFrame(x = xrange))

plot!(xrange, yrange, lw = 3, color = :blue)

File: Python-files/Getting Data and Linear Models.py
# Reading data from CSV
import pandas as pd

# Read a CSV file into a DataFrame
data = pd.read_csv("data.csv")

# Display the first few rows
print(data.head())

# Check data structure and info
print(data.info())

# Get summary statistics for numerical columns
print(data.describe())

# Fitting a linear Model
import statsmodels.api as sm

# Define independent (x) and dependent (y) variables
X = data['x']
y = data['y']

# Add a constant term (intercept)
X = sm.add_constant(X)

# Fit linear regression model
model = sm.OLS(y, X).fit()

# Display model summary
print(model.summary())

# Predict for new x values
new_x = pd.DataFrame({'x': [2, 5, 8]})
new_x = sm.add_constant(new_x)

predictions = model.predict(new_x)
print(predictions)

# Plotting Regression Line
import matplotlib.pyplot as plt

plt.scatter(data['x'], data['y'], label='Data', color='gray')
plt.plot(data['x'], model.predict(X), color='blue', label='Regression Line')
plt.xlabel('x')
plt.ylabel('y')
plt.legend()
plt.show()
